### PR TITLE
[dart] [cpd] Added CPD support for Dart

### PR DESCRIPTION
**Pull request by maikelsteneker**
Before submitting a PR, please check that:
 - [x] The PR is submitted against `master`. The PMD team will merge back to support branches as needed.
 - [x] `./mvnw clean verify` passes. This will [build](https://github.com/pmd/pmd/blob/master/BUILDING.md) and test PMD, execute PMD and checkstyle rules. [Check this for more info](https://github.com/pmd/pmd/blob/master/CONTRIBUTING.md#code-style)

**PR Description:**
This is a pull request for the addition of Dart support for CPD. This does not include full PMD support, so it can only be used to check for code duplication.

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">No java rules are changed!</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pmd-dart/src/main/java/net/sourceforge/pmd/cpd/DartTokenizer.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    protected AntlrTokenFilter getTokenFilter(final AntlrTokenManager tokenManager) {
+        return new DartTokenFilter(tokenManager);
+    }
+
+    /**
+     * The {@link DartTokenFilter} extends the {@link AntlrTokenFilter} to discard
+     * Dart-specific tokens.
+     * <p>
+     * By default, it discards package and import statements, and
+     * enables annotation-based CPD suppression.
```
Comment (jsotuyod) [inline pmd-dart/src/main/java/net/sourceforge/pmd/cpd/DartTokenizer.java]: it's actually *comment-based* CPD suppression

**Comment by jsotuyod** [comment]
@maikelsteneker merged! this will be part of PMD 6.14.0.

Thanks for the effort!

**Comment by maikelsteneker** [comment]
Glad to hear that! Thank you as well.

**Comment by matifraga** [comment]
I'm happy to see how easily we can add languages now!! @maikelsteneker do you have any feedback regarding the documentation we provide to add new languages using ANTLR?

**Comment by maikelsteneker** [comment]
@matifraga I only discovered the documentation after adding some languages through experimentation, so I may be blind to the details that I already understand now. But to me, the steps are very clear and this is indeed the process I follow myself.

**Pull request merged**